src/plugins/data/public/ui/search_bar/search_bar.tsx

```diff
@@ -521,6 +548,7 @@ class SearchBarUI extends Component<SearchBarProps, State> {
     return (
       <div className={className} data-test-subj="globalQueryBar">
         {queryBar}
+        {searchBarExtensions}
```

Comment:
along with: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6894/files#r1626586752, then we just need to do the work in the query editor